### PR TITLE
Shruthi test

### PR DESCRIPTION
**Pull request by ShruthiNiranjan**

### CODE REVIEW

File: src/test/java/SalesforceModules/InvoiceUtil.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+     * @param account the account name
+     */
+    public static void selectAccountPayments(String account, String logStep) throws InterruptedException {
+        boolean flag = false;
+        KeywordUtil.waitForVisible(ServiceSupportObject.searchAccounts_quote_ss);
+        KeywordUtil.click(ServiceSupportObject.searchAccounts_quote_ss, logStep);
+        KeywordUtil.inputText(ServiceSupportObject.searchAccounts_quote_ss, "Bill", logStep);
+        KeywordUtil.delay(5000);
+        Thread.sleep(5000);
+        String accname = "//lightning-base-combobox-formatted-text[contains(@title,'Bill')]";
```
Comment (snaphray) [inline src/test/java/SalesforceModules/InvoiceUtil.java]: do we need this xpath here as it is not dynamic?

**Review by snaphray — CHANGES_REQUESTED** [review]
Comments provided

File: src/test/java/SalesforceModules/CampaignUtil.java [hunk 2]
```diff
@@ -1324,33 +1323,94 @@ public static void clickAddtoCampaign(String logStep) throws InterruptedExceptio
...
+        KeywordUtil.delay(3000);
+
+        KeywordUtil.scrollingToElementofAPage(ServiceSupportObject.checkbox_Orders_ss,"scrolling to the order checkbox");
+        KeywordUtil.clickUsingAction(ServiceSupportObject.checkbox_Orders_ss, "Order button clicked");
+        KeywordUtil.delay(8000);
+        KeywordUtil.click(CampaignObject.saveEditQuote,"click on save button present on edit quote");
+        KeywordUtil.delay(8000);
+        KeywordUtil.waitForClickable(ServiceSupportObject.openOrder);
+        WebElement order = KeywordUtil.getDriver().findElement(ServiceSupportObject.openOrder);
+        JavascriptExecutor executor = (JavascriptExecutor)KeywordUtil.getDriver();
```
Comment (snaphray) [inline src/test/java/SalesforceModules/CampaignUtil.java:1361]: why are we not using clickJS method here?

File: src/test/java/SalesforceModules/CampaignUtil.java [hunk 3]
```diff
@@ -1324,33 +1323,94 @@ public static void clickAddtoCampaign(String logStep) throws InterruptedExceptio
...
+     * Activate order in marketing module
+     *
+     *
+     */
+    public static void activateOrder_ss() throws InterruptedException {
+        KeywordUtil.delay(3000);
+        KeywordUtil.waitForClickable(CampaignObject.viewOrder);
+        WebElement order = KeywordUtil.getDriver().findElement(CampaignObject.viewOrder);
+        JavascriptExecutor executor_ss = (JavascriptExecutor)KeywordUtil.getDriver();
+        executor_ss.executeScript("arguments[0].click();", order);
```
Comment (snaphray) [inline src/test/java/SalesforceModules/CampaignUtil.java:1398]: Same comment as above?.

File: src/test/java/SalesforceModules/CampaignUtil.java [hunk 4]
```diff
@@ -1324,33 +1323,94 @@ public static void clickAddtoCampaign(String logStep) throws InterruptedExceptio
...
+        KeywordUtil.delay(300);
+        Thread.sleep(3000);
+        KeywordUtil.waitForVisible(QuoteObject.activatedTab);
+        WebElement quote = KeywordUtil.getDriver().findElement(QuoteObject.activatedTab);
+        JavascriptExecutor executor = (JavascriptExecutor)KeywordUtil.getDriver();
+        executor.executeScript("arguments[0].click();", quote);
```
Comment (snaphray) [inline src/test/java/SalesforceModules/CampaignUtil.java:1404]: Same as above

File: src/test/java/SalesforceModules/CampaignUtil.java [hunk 5]
```diff
@@ -1324,33 +1323,94 @@ public static void clickAddtoCampaign(String logStep) throws InterruptedExceptio
...
         try {
-            KeywordUtil.scrollingToElementofAPage(CampaignObject.editQuoteButton,"scroll to the edit quote button");
-            KeywordUtil.click(CampaignObject.editQuoteButton, "click on the edit quote button");
-
+            KeywordUtil.waitForVisible(QuoteObject.markCurrentStatus);
+            KeywordUtil.click(QuoteObject.markCurrentStatus, "Activated status Marked");
         }catch (Exception e){
-
+            WebElement markStatus = KeywordUtil.getDriver().findElement(QuoteObject.markCurrentStatus);
+            executor.executeScript("arguments[0].click();", markStatus);
```
Comment (snaphray) [inline src/test/java/SalesforceModules/CampaignUtil.java:1410]: Same as above

File: src/test/java/SalesforceModules/InvoiceUtil.java [hunk 6]
```diff
@@ -229,7 +229,68 @@ public static void paymentsBilling(String account,String amount,String paymentty
...
+     * @param optionalSKU the optionalSKU
+     */
+    public static void selectOptionalSKU(String optionalSKU, String logStep) throws InterruptedException {
+        boolean flag = false;
+        KeywordUtil.waitForVisible(InvoiceObject.selectOptionalSKU);
+        KeywordUtil.click(InvoiceObject.selectOptionalSKU, logStep);
+        KeywordUtil.inputText(InvoiceObject.selectOptionalSKU, "Sam", logStep);
+        KeywordUtil.delay(5000);
+        Thread.sleep(5000);
+        String accname = "//lightning-base-combobox-formatted-text[contains(@title,'Samsung')]";
```
Comment (snaphray) [inline src/test/java/SalesforceModules/InvoiceUtil.java]: Why we are using the xpath here directly if it is not dynamic?

File: src/test/java/SalesforceModules/SalesForceBillingUtil.java [hunk 7]
```diff
@@ -337,6 +337,135 @@ public static void createNewContactOfContacts(String salutation, String name, St
...
+        CasesUtil.clickSaveButton("Click on save button");
+    }
+
+    /**
+     * Input the webCompany
+     * @param Company the Company
+     * @param logStep the logStep
+     * @throws InterruptedException
+     */
+    public static void webCompany(String Company, String logStep) throws InterruptedException {
```
Comment (snaphray) [inline src/test/java/SalesforceModules/SalesForceBillingUtil.java]: Argument name should start from small letter

File: src/test/java/pageobjects/SalesforceObjects/CampaignObject.java [hunk 8]
```diff
@@ -166,11 +166,19 @@ public class CampaignObject {
 
     public static By editQuoteButton=By.xpath("//a[@title=\"Edit\"]");
 
+    public static By editQuote=By.xpath("//button[@title=\"Edit Status\"]");
+    public static By editQuoteDraft=By.xpath("//button[@data-value=\"Draft\"]");
+
+    public static By editQuoteAgain=By.xpath("(//button[@name=\"Edit\"])[2]");
+    public static By saveEditQuote=By.xpath("//button[@name='SaveEdit']");
+   // public static By viewOrder=By.xpath("//table[@role=\"grid\"]//tbody//tr[@class=\"slds-hint-parent\"]//th[@data-label=\"Order Number\"]");
```
Comment (snaphray) [inline src/test/java/pageobjects/SalesforceObjects/CampaignObject.java]: Remove the commented line

File: src/test/java/step_definitions/SalesForceMarketing.java [hunk 9]
```diff
@@ -286,9 +286,17 @@ public static void changeLeadStatus() {
         }
     }
 
-    @And("Create an order")
-    public void createOrder(){
-        CampaignUtil.createOrder();
+    @And("Create an order and activate in marketing crm")
+    public void create_order_And_Activate() {
```
Comment (snaphray) [inline src/test/java/step_definitions/SalesForceMarketing.java]: Please follow the same name pattern throughout the file, it should not include the '_'

File: src/test/java/pageobjects/SalesforceObjects/CampaignObject.java (hunk 8 above)
Comment (ShruthiNiranjan) [reply src/test/java/pageobjects/SalesforceObjects/CampaignObject.java]: done

File: src/test/java/SalesforceModules/SalesForceBillingUtil.java (hunk 7 above)
Comment (ShruthiNiranjan) [reply src/test/java/SalesforceModules/SalesForceBillingUtil.java]: done

File: src/test/java/step_definitions/SalesForceMarketing.java (hunk 9 above)
Comment (ShruthiNiranjan) [reply src/test/java/step_definitions/SalesForceMarketing.java]: done

File: src/test/java/SalesforceModules/InvoiceUtil.java (hunk 6 above)
Comment (ShruthiNiranjan) [reply src/test/java/SalesforceModules/InvoiceUtil.java]: done

File: src/test/java/SalesforceModules/InvoiceUtil.java (hunk 1 above)
Comment (ShruthiNiranjan) [reply src/test/java/SalesforceModules/InvoiceUtil.java]: done

File: src/test/java/SalesforceModules/CampaignUtil.java (hunk 4 above)
Comment (ShruthiNiranjan) [reply src/test/java/SalesforceModules/CampaignUtil.java:1404]: Will keep this same

**Review by snaphray — APPROVED** [review]
Code looks fine now

**Pull request merged**